src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Node/JsonArrayConverter.cs

```diff
@@ -34,10 +34,29 @@ public override void Write(Utf8JsonWriter writer, JsonArray? value, JsonSerializ
             }
         }
 
-        public static JsonArray ReadList(ref Utf8JsonReader reader, JsonNodeOptions? options = null)
+        public static JsonArray ReadList(ref Utf8JsonReader reader, JsonSerializerOptions  options)
         {
-            JsonElement jElement = JsonElement.ParseValue(ref reader);
-            return new JsonArray(jElement, options);
+            JsonNodeOptions nodeOptions = options.GetNodeOptions();
+
+            if (options.AllowDuplicateProperties)
+            {
+                JsonElement jElement = JsonElement.ParseValue(ref reader);
+                return new JsonArray(jElement, nodeOptions);
+            }
+            else if (options.PropertyNameCaseInsensitive)
+            {
+                // Do duplicate detection by expanding the JsonArray eagerly
+                JsonElement jElement = JsonElement.ParseValue(ref reader);
```

Comment:
Even if we don't remove the intermediate `JsonElement` from `JsonNode` deserialization, I think this branch is particularly wasteful since we're copying data into an unrented `JsonElement` only to fully replace it with an eagerly evaluated `JsonNode`. Shouldn't this just map the `Utf8JsonReader` directly to a `JsonNode`?

Reply:
Removed the JsonElement+Traversal.